### PR TITLE
Properly set col names for shard rels that citus_extradata_container points to

### LINKED ISSUE #5932: Renaming columns of a table breaks queries on depending views
I can repro this at least from 10.0, likely to be even from earlier

```SQL
create table test_t(a int);
SELECT create_distributed_table('test_t', 'a');
CREATE VIEW v1 AS SELECT * FROM test_t;
CREATE VIEW v2(k) AS SELECT * FROM test_t;

alter table test_t  rename COLUMN a TO b;
table v1;
ERROR:  column "a" does not exist
CONTEXT:  while executing command on localhost:9705


 table v2;
ERROR:  column "a" does not exist
CONTEXT:  while executing command on localhost:9704
Time: 24.497 ms

```

And, the view query looks sane:
```SQL
\d+ v1
                              View "public.v1"
┌────────┬─────────┬───────────┬──────────┬─────────┬─────────┬─────────────┐
│ Column │  Type   │ Collation │ Nullable │ Default │ Storage │ Description │
├────────┼─────────┼───────────┼──────────┼─────────┼─────────┼─────────────┤
│ a      │ integer │           │          │         │ plain   │             │
└────────┴─────────┴───────────┴──────────┴─────────┴─────────┴─────────────┘
View definition:
 SELECT test_t.b AS a
   FROM test_t;
```

### PR DESCRIPTION
**Pull request by onurctirtir**
Deparser function set_relation_column_names() knows that it needs to
re-evaluate column names based on relation's tuple descriptor when
the rte belongs to a relation (RTE_RELATION).

However before this commit, it didn't know about the fact that citus
might wrap such an rte with an rte that points to
citus_extradata_container() placeholder.

And because of this, it was simply taking the column aliases
(e.g., "bar" in "foo AS bar") into the account and this might result in
an incorrectly deparsed query as in below case:

* Say, if we had view based on following query:
  ```sql
  SELECT a FROM table;
  ```
* And if we rename column "a" to "b", the view query normally becomes:
  ```sql
  SELECT b AS a FROM table;
  ```
* So before this commit, deparsing a query based on that view was
  resulting in such a query due to deparsing based on the column aliases,
  which is not correct:
  ```sql
  SELECT a FROM table;
  ```

Fixes #5932.

DESCRIPTION: Fixes a bug that might cause failing to query the views based on tables that have renamed columns

### CODE REVIEW

File: src/backend/distributed/deparser/ruleutils_13.c [hunk 1]
```diff
@@ -1003,7 +1003,8 @@ set_relation_column_names(deparse_namespace *dpns, RangeTblEntry *rte,
 	 * get_rte_attribute_name, except that it's important to disregard dropped
 	 * columns.  We put NULL into the array for a dropped column.
 	 */
-	if (rte->rtekind == RTE_RELATION)
+	if (rte->rtekind == RTE_RELATION ||
+        GetRangeTblKind(rte) == CITUS_RTE_SHARD)
```
Comment (onurctirtir) [inline src/backend/distributed/deparser/ruleutils_13.c:1007]: As in other deparser functions, we could do;
```suggestion
	int rtekind = GetRangeTblKind(rte);
	if (rtekind == CITUS_RTE_RELATION ||
        rtekind == CITUS_RTE_SHARD)
```

But effectively both are checking for the same thing.

**Review by naisila — COMMENTED** [review]
On a first look, this seems correct to me. It's similar to one of my recent PRs #6400 . However I am not approving since I don't have time to check it in detail right now. If the state remains the same tomorrow morning, I will review more comprehensively then. Thanks!

File: src/backend/distributed/deparser/ruleutils_13.c (hunk 1 above)
Comment (naisila) [reply src/backend/distributed/deparser/ruleutils_13.c:1007]: I searched for `GetRangeTblKind` instances in ruleutils_15.c, and in all, except one instance, we directly do comparisons from `GetRangeTblKind` function. So, I vote for the current choice, not the comment's suggestion.

**Review by naisila — APPROVED** [review]
LGTM 🚀

File: src/backend/distributed/deparser/ruleutils_13.c (hunk 1 above)
Comment (onurctirtir) [reply src/backend/distributed/deparser/ruleutils_13.c:1007]: Besides doing the comparison with GetRangeTblKind(), I'm also wondering whether we should compare `rte->rtekind with RTE_RELATION`, or `GetRangeTblKind(rte) with CITUS_RTE_RELATION`.

Both should mean the same, latter one sounds slightly more intuitive.

Comment (naisila) [reply src/backend/distributed/deparser/ruleutils_13.c:1007]: Hmm now I see your point. That one exception instance is actually the only instance that is similar to this one: where we put both `RTE_RELATION` and `CITUS_RTE_SHARD` under the same umbrella. In that case, I am changing my vote :D up to you to decide.

**Pull request merged**